### DISCUSSION TITLE
Refetch static data

### DISCUSSION THREAD

**Discussion by RazvanGorea** [vercel/next.js#37701] [Ideas]
I wish next.js had a way to refetch (or invalidate) the page's static data. Once page data is loaded, the single way to update it is full refresh, which is inconvenient. One workaround I found is using static generation with something like SWR on the client-side, but why hit the API with unnecessary requests when data is already cached?. In my opinion absence of this feature is keeping new 'on-demand ISR' from being a very powerful and efficient tool.

**Comment by icyJoseph** [comment]
I have a solution which works only once, using `router.replace`. Odd that because it is a static page, it only fetches a fresh json file once. 

It is almost the same as refreshing the page, but with replace instead, and while also keeping state.

For the record, it looks like this. Remember, **it works only once!** This could probably be enhanced, by the framework, to fetch again as needed. Just like there's a `shallow` option (`router.replace('/foo', undefined, {shallow: true})`), perhaps there could be a `isrJSONFetch` option.

```ts
 const revalidate = async () => {
    try {
      await fetch("/api/revalidate?secret=foobar");
      await router.replace(router.asPath);
    } catch (e) {
      if (e instanceof Error) {
        return console.error(e);
      }
      return console.error("Unexpected error", e);
    }
  };
```
